src/server.rs

```diff
@@ -271,6 +310,59 @@ impl Server {
         }
     }
 
+    /// Server browser related query packet processing for shared socket mode.  These are used
+    /// when you pass STEAMGAMESERVER_QUERY_PORT_SHARED as the query port to SteamGameServer_Init.
+    /// IP address and port are in host order, i.e 127.0.0.1 == 0x7f000001
+    ///
+    /// Source games use this to simplify the job of the server admins, so they
+    /// don't have to open up more ports on their firewalls.
+    ///
+    /// Call this when a packet that starts with 0xFFFFFFFF comes in on the shared socket.
+    pub fn handle_incoming_packet(&self, data: &mut [u8], addr: SocketAddrV4) -> bool {
```

Comment:
`data`  need not be a mutable reference
```suggestion
    pub fn handle_incoming_packet(&self, data: &[u8], addr: SocketAddrV4) -> bool {
```